### PR TITLE
MM-52173 Add setting to delay channel autocomplete

### PR DESCRIPTION
**Pull request by hmhealey**
#### Summary
The channel autocomplete can be a bit obtrusive in certain cultures such as Korean where tildes are used more regularly in messages, so we've added a setting to make it require typing two letters after the tilde before the autocomplete triggers.

I didn't see any place to put minimum version in here, but it'll be available in ~~Mattermost 8.0 and whatever the corresponding mobile app version is~~ the May release

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-52173

#### Related Pull Requests
https://github.com/mattermost/mattermost-server/pull/22952
https://github.com/mattermost/mattermost-mobile/pull/7297

### CODE REVIEW

**Comment by itao** [comment]
Does this require v8.0, which is tentatively targeting June, or is there opportunity to release sooner in May?

**Comment by hmhealey** [comment]
Whoops. I thought 8.0 was the next release. I should really go back to using release months instead of version numbers 😅

**Review by larkox — APPROVED** [review]

File: source/configure/experimental-configuration-settings.rst [hunk 1]
```diff
@@ -660,6 +660,19 @@ See the `Important Upgrade Notes <https://docs.mattermost.com/upgrade/important-
 | This feature's ``config.json`` setting is ``"PatchPluginsReactDOM": false`` with options ``true`` and ``false``. |
 +------------------------------------------------------------------------------------------------------------------+
 
+Delay channel autocomplete
+~~~~~~~~~~~~~~
```
Comment (cwarnermm) [inline source/configure/experimental-configuration-settings.rst:664]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: source/configure/experimental-configuration-settings.rst [hunk 2]
```diff
@@ -660,6 +660,19 @@ See the `Important Upgrade Notes <https://docs.mattermost.com/upgrade/important-
...
+
+This setting controls whether or not the channel link autocomplete when making a post triggers immediately when after a tilde is typed.
+
+**True**: The autocomplete appears after the user types a tilde followed by two or more letters. For example, typing `~to` will show the autocomplete but typing `~` will not.
+
+**False**: **(Default)** The autocomplete appears immediately after the user types a tilde. For example, typing `~` will show the autocomplete.
+
++-----------------------------------------------------------------------------------------------------------------------+
+| This feature's ``config.json`` setting is ``"DelayChannelAutocomplete": false`` with options ``true`` and ``false``. |
++------------------------------------------------------------------------------------------------------------------+
```
Comment (cwarnermm) [inline source/configure/experimental-configuration-settings.rst:674]:
```suggestion
+-----------------------------------------------------------------------------------------------------------------------+
```

File: source/configure/experimental-configuration-settings.rst [hunk 3]
```diff
@@ -660,6 +660,19 @@ See the `Important Upgrade Notes <https://docs.mattermost.com/upgrade/important-
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This setting controls whether or not the channel link autocomplete when making a post triggers immediately when after a tilde is typed.
+
+**True**: The autocomplete appears after the user types a tilde followed by two or more letters. For example, typing `~to` will show the autocomplete but typing `~` will not.
+
+**False**: **(Default)** The autocomplete appears immediately after the user types a tilde. For example, typing `~` will show the autocomplete.
+
++-----------------------------------------------------------------------------------------------------------------------+
+| This feature's ``config.json`` setting is ``"DelayChannelAutocomplete": false`` with options ``true`` and ``false``. |
```
Comment (cwarnermm) [inline source/configure/experimental-configuration-settings.rst:673]:
```suggestion
| This feature's ``config.json`` setting is ``"DelayChannelAutocomplete": false`` with options ``true`` and ``false``.  |
```

**Review by cwarnermm — APPROVED** [review]
Thanks, @hmhealey!!

**Review by wiersgallak — APPROVED** [review]

**Comment by cwarnermm** [comment]
@hmhealey - Can we merge this docs PR, or is it waiting on an upcoming release of mobile app?

**Comment by hmhealey** [comment]
The setting won't exist in on-prem versions of MM until 8.0 comes out, and I'm not sure if they're on Cloud yet. I'd be fine merging them whenever, but I don't know what our standard practice is when it comes to new settings, especially because there's been some unusually long delays with this release

**Comment by cwarnermm** [comment]
@hmhealey - I'm going to close this PR in favour of https://github.com/mattermost/docs/pull/6461/.

**Pull request closed without merging**